platform_plugin_aspects/views.py

```diff
@@ -0,0 +1,136 @@
+
+        def has_object_permission(self, request, view, obj):
+            """
+            Return False for security; mock this out during tests.
+            """
+            return False
+
+
+# Course fields:
+# * course_id: CourseKey; required by IsCourseStaffInstructor
+# * display_name: str; optionally fetched from CourseOverview
+Course = namedtuple("Course", ["course_id", "display_name"])
+
+
+class SupersetView(GenericAPIView):
+    """
+    Superset-related endpoints provided by the aspects platform plugin.
+    """
+
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (
+        permissions.IsAuthenticated,
+        IsStaffOrReadOnly | IsCourseStaffInstructor,
+    )
+
+    lookup_field = "course_id"
+
+    @property
+    def allowed_methods(self):
+        """
+        Only POST is allowed for this view.
+        """
+        return ["post", "options", "head"]
+
+    def get_object(self):
+        """
+        Return a Course-like object for the requested course_id.
+        """
+        course_id = self.kwargs.get(self.lookup_field, "")
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError as exc:
+            raise NotFound(
+                _("Invalid course id: '{course_id}'").format(course_id=course_id)
+            ) from exc
+
+        # Fetch the CourseOverview (if we're running in edx-platform)
+        display_name = ""
+        CourseOverview = get_model("course_overviews")
+        if CourseOverview:
+            try:
+                course_overview = CourseOverview.objects.get(id=course_key)
+                display_name = course_overview.display_name
+            except CourseOverview.DoesNotExist as exc:
+                raise NotFound(
+                    _("Course not found: '{course_id}'").format(course_id=course_id)
+                ) from exc
+
+        course = Course(course_id=course_key, display_name=display_name)
+
+        # May raise a permission denied
+        self.check_object_permissions(self.request, course)
+
+        return course
+
+    def post(self, request, *args, **kwargs):
+        """
+        Return a guest token for accessing the Superset API.
+        """
+        course = self.get_object()
+
+        built_in_filters = [
+            f"org = '{course.course_id.org}'",
+            f"course_name = '{course.display_name}'",
+            f"course_run = '{course.course_id.run}'",
+        ]
+        dashboards = settings.ASPECTS_INSTRUCTOR_DASHBOARDS
+        extra_filters_format = settings.SUPERSET_EXTRA_FILTERS_FORMAT
+
+        try:
+            guest_token = generate_guest_token(
+                user=request.user,
+                course=course,
+                dashboards=dashboards,
+                filters=built_in_filters + extra_filters_format,
+            )
+        except ImproperlyConfigured as exc:
+            raise APIException() from exc
+
+        return Response({"guestToken": guest_token})
```

Comment:
Should this be a JSON response?

Reply:
I think DRF returns JSON by default.. but I'll check.